Complex.mo

```diff
@@ -166,21 +166,35 @@ operator record Complex "Complex number with overloaded operators"
 </html>"));
   end '/';
 
-  encapsulated operator function '^' "Complex power of complex number"
-    import Complex;
-    input Complex c1 "Complex number";
-    input Complex c2 "Complex exponent";
-    output Complex c3 "= c1^c2";
-  protected
-    Real lnz=0.5*log(c1.re*c1.re + c1.im*c1.im);
-    Real phi=atan2(c1.im, c1.re);
-    Real re=lnz*c2.re - phi*c2.im;
-    Real im=lnz*c2.im + phi*c2.re;
-  algorithm
-    c3 := Complex(exp(re)*cos(im), exp(re)*sin(im));
-    annotation(Inline=true, smoothOrder=100, Documentation(info="<html>
+  encapsulated operator '^' "Power"
+    function complexPower "Complex power of complex number"
+      import Complex;
+      input Complex c1 "Complex number";
+      input Complex c2 "Complex exponent";
+      output Complex c3 "= c1^c2";
+    protected
+      Real lnz=0.5*log(c1.re*c1.re + c1.im*c1.im);
+      Real phi=atan2(c1.im, c1.re);
+      Real re=lnz*c2.re - phi*c2.im;
+      Real im=lnz*c2.im + phi*c2.re;
+    algorithm
+      c3 := Complex(exp(re)*cos(im), exp(re)*sin(im));
+      annotation(Inline=true, smoothOrder=100, Documentation(info="<html>
 <p>This function returns the given Complex numbers c1 to the power of the Complex number c2.</p>
 </html>"));
+    end complexPower;
+    function integerPower "Integer power of complex number"
+      import Complex;
+      input Complex c1 "Complex number";
+      input Integer c2 "Integer exponent";
+      output Complex c3 "= c1^c2";
+    algorithm
+      c3 := if c2==0 then Complex(1,0) else Complex.'^'.complexPower(c1,Complex(c2,0));
```

Comment:
Two comments on `Complex(c2,0)`
* It uses the fromReal ctor which is an implicit cast from Integer to Real, which is OK is guess.
* The second argument can be skipped.